joinmarket/wallet.py

```diff
@@ -177,11 +177,16 @@ def read_wallet_file_data(self, filename, pwd=None):
                 self.index_cache += [[0,0]] * (
                     self.max_mix_depth - len(self.index_cache))
         decrypted = False
+        trieddefault = False
         while not decrypted:
-            if pwd:
-                password = pwd
+            if trieddefault:
+                password = ''
+                trieddefault = True
```

Comment:
I can't see the logic of this (lines 180-184, plus 204-5); `trieddefault` is set to False before the decryption-attempting loop, and the password is set to '' if `trieddefault` is True; but it is not set to True anywhere except within that condition where it has already evaluated as True. I'm guessing what you intended is, line 182 to read `if not trieddefault`, thus to enforce an attempt to unlock with a null password once only.